### PR TITLE
Chess tutorial

### PR DESCRIPTION
**Pull request by rkdud007**
use [dojo-chess](https://github.com/rkdud007/chess-dojo/tree/tutorialv2) project a full tutorial project in dojo book
- [X] [Building Onchain Chess Game
  - [X] [0. Setup]
  - [X] [1. Initiate]
  - [X] [2. Move]
  - [X] [3. Check legal]
  - [X] [4. Test Chess]

### CODE REVIEW

**Comment by rkdud007** [comment]
Can review the tutorial? cc @ponderingdemocritus

File: src/tutorial/onchain-chess/0-setup.md [hunk 1]
```diff
@@ -0,0 +1,385 @@
+# 0. Setup
+
+_Before you start, I recommend you to follow `hello-dojo` chaper to get a brief understanding of Dojo game._
```
Comment (ponderingdemocritus) [inline src/tutorial/onchain-chess/0-setup.md:3]: add link here

File: src/tutorial/onchain-chess/0-setup.md [hunk 2]
```diff
@@ -0,0 +1,385 @@
...
+
+Open your empty game folder and open the terminal.
+
+```sh
+sozo init
+```
+
+## setup
+
+The project comes with a lot of boilerplate codes. Clear them all. Make sure both `components.cairo` and `systems.cairo` files are empty.
```
Comment (ponderingdemocritus) [inline src/tutorial/onchain-chess/0-setup.md:17]:
```suggestion
The project comes with a lot of boilerplate codes. Clear it all. Make sure both `components.cairo` and `systems.cairo` files are empty.
```

File: src/tutorial/onchain-chess/0-setup.md [hunk 3]
```diff
@@ -0,0 +1,385 @@
...
+
+In `lib.cairo`, retain only:
+
+```rust,ignore
+mod components;
+mod systems;
+```
+
+Build your initial setup project using `sozo build`.
```
Comment (ponderingdemocritus) [inline src/tutorial/onchain-chess/0-setup.md:26]:
```suggestion
Compile your project with:

```
sozo build
```
```

File: src/tutorial/onchain-chess/0-setup.md [hunk 4]
```diff
@@ -0,0 +1,385 @@
...
+    BlackPawn,
+    BlackKnight,
+    BlackBishop,
+    BlackRook,
+    BlackQueen,
+    BlackKing,
+}
+```
+
+First, add this basic component to `components.cairo` file. If you are not familar with component syntax in Dojo engine, go back to this [chapter](../../framework/cairo/components.md).
```
Comment (ponderingdemocritus) [inline src/tutorial/onchain-chess/0-setup.md:60]: this could be better above the code I feel

File: src/tutorial/onchain-chess/0-setup.md [hunk 5]
```diff
@@ -0,0 +1,385 @@
...
+
+## Basic systems
+
+Starting from the next chapter, you will implement `initiate` and `move` systems one by each chapter. Let's make each system per file so that it can be more modularized.
```
Comment (ponderingdemocritus) [inline src/tutorial/onchain-chess/0-setup.md:64]:
```suggestion
Starting from the next chapter, you will implement `initiate` and `move` systems one in each chapter. Let's create each system in its own file for better modularity.
```

File: src/tutorial/onchain-chess/0-setup.md [hunk 6]
```diff
@@ -0,0 +1,385 @@
...
+
+Create `systems` folder at `src`. Create `initiate.cairo`and `move.cairo` two files inside the folder. Each file should contain a basic system structure.
```
Comment (ponderingdemocritus) [inline src/tutorial/onchain-chess/0-setup.md:66]:
```suggestion
Create a `systems` folder at `src`. Create `initiate.cairo`and `move.cairo` two files inside the folder. Each file should contain a basic system structure.
```

File: src/tutorial/onchain-chess/0-setup.md [hunk 7]
```diff
@@ -0,0 +1,385 @@
...
+
+For example, `initiate.cairo` would be looks like this:
```
Comment (ponderingdemocritus) [inline src/tutorial/onchain-chess/0-setup.md:68]:
```suggestion
For example, `initiate.cairo` would look like this:
```

File: src/tutorial/onchain-chess/0-setup.md [hunk 8]
```diff
@@ -0,0 +1,385 @@
...
+
+and in `systems.cairo` we will use initiate_system like this:
+
+```rust
+mod initiate;
+
+use initiate::initiate_system;
+```
+
+Do the same with the other three systems. And make update `systems.cairo` like :
```
Comment (ponderingdemocritus) [inline src/tutorial/onchain-chess/0-setup.md:85]:
```suggestion
Do the same with the other three systems. And make update `systems.cairo` like:
```

**Pull request closed without merging**